Add iOS target for demo-compose-app

siarhei-luskanau opened the pull request:
<!--
Thank you for opening a pull request!

Please add a brief description of the proposed change here.
Also, please tick the appropriate points in the checklist below.
-->

## Motivation and Context
<!-- Why is this change needed? What problem does it solve? -->

## Breaking Changes
<!-- Will users need to update their code or configurations? -->

---

#### Type of the changes
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [ ] Tests improvement
- [x] Refactoring

#### Checklist
- [x] The pull request has a description of the proposed change
- [x] I read the [Contributing Guidelines](https://github.com/JetBrains/koog/blob/main/CONTRIBUTING.md) before opening the pull request
- [x] The pull request uses **`develop`** as the base branch
- [ ] Tests for the changes have been added
- [ ] All new and existing tests passed

##### Additional steps for pull requests adding a new feature
- [ ] An issue describing the proposed change exists
- [ ] The pull request includes a link to the issue
- [ ] The change was discussed and approved in the issue
- [x] Docs have been added / updated

siarhei-luskanau commented:
iOS target has network issue. Looks like CIO engine as used instead of Darwin engine for iOS target.

<img width="1179" height="2556" alt="Simulator Screenshot - iPhone 16 - 2025-09-10 at 13 06 32" src="https://github.com/user-attachments/assets/d5ba1b6f-09fa-4b45-af89-bda153d5d14c" />

```
ai.koog.agents.core.exception.UnexpectedServerException: TLS sessions are not supported on Native platform.
    at 0   Koog Demo App.debug.dylib           0x10a07f00f        kfun:kotlin.Throwable#<init>(kotlin.String?){} + 99 
    at 1   Koog Demo App.debug.dylib           0x10a078cbb        kfun:kotlin.Exception#<init>(kotlin.String?){} + 95 
    at 2   Koog Demo App.debug.dylib           0x10a078e8b        kfun:kotlin.RuntimeException#<init>(kotlin.String?){} + 95 
    at 3   Koog Demo App.debug.dylib           0x10b952703        kfun:ai.koog.agents.core.exception.AgentRuntimeException#<init>(kotlin.String){} + 95 
    at 4   Koog Demo App.debug.dylib           0x10b952777        kfun:ai.koog.agents.core.exception.AgentEngineException#<init>(kotlin.String){} + 95 
    at 5   Koog Demo App.debug.dylib           0x10b9527eb        kfun:ai.koog.agents.core.exception.UnexpectedServerException#<init>(kotlin.String){} + 95 
```

kpavlov commented:
@siarhei-luskanau , as far as I remember, "checks" doesn't build the demo app, and demo app workflow doesn't build main project. So we should be good

siarhei-luskanau commented:
Rebased to the latest develop branch and squashed into one commit

kpavlov reviewed (approved):
Thank you, @siarhei-luskanau

siarhei-luskanau commented:
@kpavlov I have updated the PR: 

- Rebased to the latest develop branch and squashed into one commit.
- Updated koog version to "0.4.2".
- Fix Android target issue "Duplicate class io.modelcontextprotocol.kotlin.sdk.LibVersionKt" by updating the koog dependencies to "ai.koog:agents-core" and "ai.koog:prompt-executor-llms-all".

Jvm, Android, iOS target works as expected for me locally.

kpavlov reviewed (commented):
Thank you, @siarhei-luskanau 🙏🏻 
Verified, works for me too

siarhei-luskanau commented:
@kpavlov  Please keep in mind that I have two additional PR related the demo-compose-app:

-  Refactor: Move buildSrc to convention-plugin-ai #831
-  Demo app in root project #832
